src/ttsim/tt_dag_elements/piecewise_polynomial.py

```diff
@@ -47,59 +51,58 @@ class RatesOptions:
 
 
 def piecewise_polynomial(
-    x: np.ndarray,
+    x: Float[Array, " n_pp_values"],
     parameters: PiecewisePolynomialParamValue,
-    rates_multiplier: np.ndarray = 1.0,
-) -> np.ndarray:
+    xnp: ModuleType,
+    rates_multiplier: Float[Array, " n_segments"] | float = 1.0,
+) -> Float[Array, " n_pp_values"]:
     """Calculate value of the piecewise function at `x`. If the first interval begins
     at -inf the polynomial of that interval can only have slope of 0. Requesting a
     value outside of the provided thresholds will lead to undefined behaviour.
 
     Parameters
     ----------
-    x : np.ndarray
+    x:
         Array with values at which the piecewise polynomial is to be calculated.
-    thresholds : np.array
-                A one-dimensional array containing the thresholds for all intervals.
-    coefficients : np.ndarray
-            A two-dimensional array where columns are interval sections and rows
-            correspond to the coefficient of the nth polynomial.
-    intercepts : np.ndarray
-        The intercepts at the lower threshold of each interval.
-    rates_multiplier : np.ndarray
-                       Multiplier to create individual or scaled rates.
+    parameters:
```

Comment:
Maybe it's somewhere else now, but I think it might be better to explain here what the parameters look like. I mean what rates (?), thresholds and intercepts are.